File: array_variables-linear_search.py
# Description: Linear search algorithm to find the first occurrence of a value in an array

names = ['John', 'Jane', 'Jack', 'Jill', 'Jenny', 'Jerry', 'Jill', 'Jenny', 'Jenny']

# Linear search algorithm
def linear_search(arr, value):
    for index, item in enumerate(arr):
        if item == value:
            return index
    return -1

print(linear_search(names, 'Jill'))  # 3
# end of linear search algorithm

# Using enumerate to get the index and value of an array
enum = enumerate(names)
for indice, value in enum:
    print(f"{indice}, {value}, {value == 'Jill'}")

# Get the memory address of the array
print(enum)

File: arrays/array_variables-linear_search.js
let names = ['Alice', 'Bob', 'Tiff', 'Bruce', 'Alice'];

function linear_search(arr, item) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === item) {
            return i;
        }
    }
    return -1;  // if the item is not found
}

console.log(linear_search(names, 'Bruce')); // 3